qiskit/qpy/interface.py

```diff
@@ -12,16 +12,30 @@
 
 """User interface of qpy serializer."""
 
+from json import JSONEncoder, JSONDecoder
+from typing import Union, List, BinaryIO
+from collections.abc import Iterable
 import struct
 import warnings
 
 from qiskit.circuit import QuantumCircuit
+from qiskit.pulse import ScheduleBlock
 from qiskit.exceptions import QiskitError
-from qiskit.qpy import formats, common, binary_io
+from qiskit.qpy import formats, common, binary_io, type_keys
+from qiskit.qpy.exceptions import QpyError
 from qiskit.version import __version__
 
 
-def dump(circuits, file_obj, metadata_serializer=None):
+# pylint: disable=invalid-name
+QPY_SUPPORTED_TYPES = Union[QuantumCircuit, ScheduleBlock]
+
+
+def dump(
+    programs: Union[List[QPY_SUPPORTED_TYPES], QPY_SUPPORTED_TYPES],
+    file_obj: BinaryIO,
+    metadata_serializer: JSONEncoder = None,
```

Comment:
A type hinting syntax question here. This actually isn't an instance of `JSONEncoder` but a subclass that's required (this is because it's what stdlib `json` works with. Does the type hint imply an instance or will something like mypy work with a subclass being passed directly.

Reply:
(also, for strict mypy support, it should be `Optional[Type[JSONEncoder]]`, but that's somewhat beside the point)

Reply:
Thanks!